File: autumn/human-readable-number/js/log.js
function numberFormat(n) {
   let n2 = Math.log(n) / Math.log(1000);
   let n3 = Math.floor(n2);
   return (n / 1000 ** n3).toFixed(3) + ['', ' k', ' M', ' G'][n3];
}

let s = numberFormat(1234);
console.log(s);

File: autumn/human-readable-number/py/log.py
import math

def number_format(d):
   e = int(math.log10(d) / 3)
   return '%.3f' % (d / 1e3 ** e) + ('', ' k', ' M', ' G')[e]

s = number_format(9012345678)
print(s == '9.012 G')
